src/appleseed/renderer/kernel/rendering/masterrenderer.cpp

```diff
@@ -297,17 +299,27 @@ struct MasterRenderer::Impl
         try
         {
             Stopwatch<DefaultWallclockTimer> stopwatch;
-            stopwatch.start();
 
+            // Render.
+            stopwatch.start();
             result.m_status = do_render();
-
             stopwatch.measure();
             result.m_render_time = stopwatch.get_seconds();
 
-            if (result.m_status == RenderingResult::Succeeded)
-                add_render_stamp(result.m_render_time);
+            // Set the frame's render info.
+            ParamArray& render_info = m_project.get_frame()->render_info();
+            render_info.clear();
+            render_info.insert("render_time", stopwatch.get_seconds());
+
+            // Don't proceed further if rendering failed.
+            if (result.m_status != RenderingResult::Succeeded)
+                return result;
 
-            return result;
+            // Post-process.
+            stopwatch.start();
+            postprocess(result);
+            stopwatch.measure();
+            result.m_post_processing_time = stopwatch.get_seconds();
```

Comment:
Why not storing it in `render_info` like `render_time` ? 
`render_info.insert("render_time", stopwatch.get_seconds());`

Reply:
I can but it won't be used at the moment (the only use of render info is the render stamp post-processing stage, which is executed before the total post-processing time is known).